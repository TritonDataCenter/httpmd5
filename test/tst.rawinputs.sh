#
# This test exercises each of the inputs found in the same directory, starting
# with "input.{success,failure}.*".  Note that we only check the exit status
# directly in this file.  The test runner will check that stdout matches what
# we'd expect, which is most of what matters.
#

cmd="$(dirname ${BASH_SOURCE[0]})/../bin/httpmd5"
shopt -s nullglob

for file in input.success.*; do
	echo "test case: raw input file $file"
	if ! $cmd < $file 2>&1; then
		echo "unexpected failure: \"$file\""
		exit 1
	fi
	echo
done

for file in input.failure.*; do
	echo "test case: raw input file $file"
	if $cmd < $file 2>&1; then
		echo "unexpected success: \"$file\""
		exit 1
	fi
	echo
done

#
# Test a case with unexpected arguments passed.
#
echo "test case: unexpected arguments"
if "$cmd" boom 2>&1; then
	echo "unexpected success with extra arguments"
	exit 1
fi
echo

echo "all test cases run"
