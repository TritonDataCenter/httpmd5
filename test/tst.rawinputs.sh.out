test case: raw input file input.success.b64_no_newline
input string:   +8JLzHoXlHWPwTJ/z+va9g==
input encoding: base64
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.success.b64_with_newline
input string:   +8JLzHoXlHWPwTJ/z+va9g==
input encoding: base64
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.success.content_md5
input string:   +8JLzHoXlHWPwTJ/z+va9g==
input encoding: base64
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.success.headers_only
input string:   +8JLzHoXlHWPwTJ/z+va9g==
input encoding: base64
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.success.hex_no_newline
input string:   fbc24bcc7a1794758fc1327fcfebdaf6
input encoding: hex
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.success.hex_with_newline
input string:   fbc24bcc7a1794758fc1327fcfebdaf6
input encoding: hex
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.success.minfo_headers
input string:   +8JLzHoXlHWPwTJ/z+va9g==
input encoding: base64
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.success.response
input string:   +8JLzHoXlHWPwTJ/z+va9g==
input encoding: base64
base64-encoded: +8JLzHoXlHWPwTJ/z+va9g==
hex-encoded:    fbc24bcc7a1794758fc1327fcfebdaf6

test case: raw input file input.failure.badhttp
httpmd5: expected "HTTP/1" at the start of multi-line input (found "Connection: close")

test case: raw input file input.failure.response_nomd5
httpmd5: found no "content-md5" header line for multi-line input

test case: raw input file input.failure.response_nomd5_body
httpmd5: found no "content-md5" header line for multi-line input

test case: raw input file input.failure.toolong
httpmd5: found no "content-md5" header line for multi-line input (note: stopped reading after 16384 bytes)

test case: unexpected arguments
usage: httpmd5 
Convert between md5sum representations used in HTTP and those produced by 
openssl tools.

all test cases run
